### PR TITLE
SYCL-specific env and README entry

### PR DESCRIPTION
**Pull request by mlange05**
Additional encoding of the build settings in a dedicated env with a custom DPCPP install, and a short README description.

Note, it's nothing fancy but it should be enough to make it easily accessible.

### CODE REVIEW

**Comment by mlange05** [comment]
Once this is in, I think I'm happy with PR #64. @MichaelSt98 please have a look if this is ok and merge if you're happy with it.

**Review by MichaelSt98 — APPROVED** [review]
Many thanks! Looks good and I just tested it. 

The only thing to change is maybe to remove `set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Xcuda-ptxas --maxrregcount=128")`. 

On this note: without `-fsycl-early-optimizations` performance is significantly reduced! Maybe some (optimisation) flags can improve the performance even further (of course not relevant here, just to mention it).

File: arch/toolchains/ecmwf-hpc2020-intel-sycl.cmake [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+####################################################################
+
+# Additional Intel DPCPP compiler for SYCL offload
+set(CMAKE_CXX_COMPILER "/home/nams/opt/dpcpp/bin/clang++")
+
+# Initial set of flags to things going with a custom DPCPP install on AC
+set(CMAKE_CXX_FLAGS "-O3 -L/home/nams/opt/dpcpp/lib -fopenmp -lstdc++")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl-early-optimizations -fsycl")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl-targets=nvptx64-nvidia-cuda")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Xcuda-ptxas --maxrregcount=128")
```
[the remark this answers is not in the document]
Comment (mlange05) [reply arch/toolchains/ecmwf-hpc2020-intel-sycl.cmake:153]: Ok, cool, thanks for checking. I'll drop those and merge this into the main PR then.

**Comment by mlange05** [comment]
Ok, test failures seem to be due to outdated loki-config. I'll merge this and rebase on the original PR to get this ready then.

**Pull request merged**